udark/udapp/src/main/java/io/underdark/app/model/Node.java

```diff
@@ -218,18 +233,30 @@ public void transportLinkDidReceiveFrame(Transport transport, Link sender, byte[
 				}
 				break;
 			case 1: //node left the mesh notification (outside of network)
-				Long toKill = Long.parseLong(message.substring(1));
+
+				String[] messagePieces = message.split("\\|");
+
+
+				Long toKill = Long.parseLong(messagePieces[1]);
 				if(!routingTable.containsKey(toKill))break;
 				Logger.info("Will delete node " + toKill + " from routing table");
 				routingTable.remove(toKill);
 				for (Link l:
-					 links) {
+						links) {
 					if(l.getNodeId() == sender.getNodeId()) continue;
 					sendFrame(l,1,""+toKill);
 				}
 				break;
+			//actual message, could be random or typed
 			case 2:
-				break; //nothing for now....this is just random data
+				messagePieces = message.split("\\|");
```

Comment:
So we show the message if we are meant to get it....sure....but if now we need to send to the person in our routing table mapped to the destination!